### PR TITLE
special-pickup-zone

### PR DESCRIPTION
**Pull request by DwivediAman-exe**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Helpers/Utils.purs [hunk 1]
```diff
@@ -377,6 +377,15 @@ rideLabelConfig _ = [
       imageUrl : "ny_pin_check_white,",
       cancelText : "GO_TO_CANCELLATION_TITLE",
       cancelConfirmImage : "ny_ic_gotodriver_zero,"
+    },
+    {
+      label : "SpecialZonePickup",
+      backgroundColor : "#53BB6F",
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Helpers/Utils.purs:383]: add this color in colour.purs file

File: Frontend/ui-driver/src/Screens/HomeScreen/View.purs [hunk 2]
```diff
@@ -356,6 +358,46 @@ driverMapsHeaderView push state =
...
+  linearLayout
+    [ height WRAP_CONTENT
+    , width MATCH_PARENT
+    , gravity CENTER_VERTICAL
+    , padding $ Padding 12 12 16 12
+    , margin $ Margin 20 15 20 15
+    , cornerRadius 8.0
+    , stroke $ "1," <> Color.yellow900
+    , background Color.yellow600
+    , visibility if (DA.any (_ == state.props.currentStage) [RideAccepted, RideStarted, ChatWithCustomer] || not state.props.statusOnline) then GONE else VISIBLE
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/View.purs:372]: use boolToVisibility here.

File: Frontend/ui-driver/src/Screens/HomeScreen/View.purs [hunk 3]
```diff
@@ -356,6 +358,46 @@ driverMapsHeaderView push state =
...
+    , orientation HORIZONTAL
+    -- , visibility according to 150 meter flag
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/View.purs:374]: Add TODO if required.

File: Frontend/ui-driver/src/Screens/HomeScreen/View.purs [hunk 4]
```diff
@@ -356,6 +358,46 @@ driverMapsHeaderView push state =
   where
     getCarouselView visible bottomMargin = maybe ([]) (\item -> if DA.any (_ == state.props.currentStage) [RideAccepted, RideStarted, ChatWithCustomer] || visible then [] else [bannersCarousal item bottomMargin state push]) state.data.bannerData.bannerItem
 
+specialMetoPickupZone :: forall w . (Action -> Effect Unit) -> HomeScreenState -> PrestoDOM (Effect Unit) w
+specialMetoPickupZone push state =
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/View.purs:362]: rename this.

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 5]
```diff
@@ -1154,10 +1161,9 @@ activeRideDetail state (RidesInfo ride) =
   waitTimeInfo : state.data.activeRide.waitTimeInfo,
   requestedVehicleVariant : ride.requestedVehicleVariant,
   waitTimerId : state.data.activeRide.waitTimerId,
-  specialLocationTag : if isJust ride.disabilityTag then Just "Accessibility"
-                        else if isJust ride.driverGoHomeRequestId then Just "GOTO" 
-                        else if isJust ride.specialLocationTag then ride.specialLocationTag
-                        else if isSafetyRide then Just "Safety"
+  specialLocationTag :  if isJust ride.disabilityTag then Just "Accessibility"
+                        else if isJust ride.specialZonePickup then Just "SpecialZonePickup"
+                        else if isJust ride.driverGoHomeRequestId then Just "GOTO"
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:1166]: @DwivediAman-exe why is safety check removed here?

File: Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs [hunk 6]
```diff
@@ -1134,6 +1159,8 @@ getRideCompletedConfig state = let
   payerVpa = state.data.endRideData.payerVpa
   bannerConfig = autopayBannerConfig state false
   disability = state.data.endRideData.disability /= Nothing
+  specialZonePickup = state.data.endRideData.specialZonePickup /= Nothing
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs:1162]: use isNothing here.

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java [hunk 7]
```diff
@@ -403,6 +420,21 @@ public void onPageScrollStateChanged(int state) {
         }
     });
 
+    private void updateExtraChargesString(SheetAdapter.SheetViewHolder holder, SheetModel model) {
+        mainLooper.post(() -> {
+            System.out.println("zxc updateExtraChargesString" + model.getSpecialZoneExtraTip());
```
Comment (MercyQueen) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java:425]: remove logs

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java [hunk 8]
```diff
@@ -403,6 +420,21 @@ public void onPageScrollStateChanged(int state) {
...
+            String pickupChargesText = model.getCustomerTip() > 0 ?
+                    getString(R.string.includes_pickup_charges_10) + " " + getString(R.string.and) + sharedPref.getString("CURRENCY", "₹") + " " + model.getCustomerTip() + " " + getString(R.string.tip) :
+                    getString(R.string.includes_pickup_charges_10);
+            if (model.getSpecialZoneExtraTip() > 0) {
+                String pickUpChargesWithZone = pickupChargesText += " and " + sharedPref.getString("CURRENCY", "₹") + " " + model.getSpecialZoneExtraTip() + " " + "Zone pickup extra";
```
Comment (MercyQueen) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java:430]: add translations

**Review by MercyQueen — CHANGES_REQUESTED** [review]

File: Frontend/ui-common/src/Components/RideCompletedCard/View.purs [hunk 9]
```diff
@@ -107,8 +108,14 @@ topPillView config push =
...
   , padding $ Padding 16 6 16 10
   , cornerRadius 22.0
-  ][
-    textView $ [
+  ][imageView
+    [ width $ V 12
+    , height $ V 12
+    , margin $ Margin 0 2 6 0
+    , visibility $ boolToVisibility (config.topCard.topPill.icon /= Nothing)
+    , imageWithFallback $ fetchImage FF_COMMON_ASSET $ fromMaybe "" config.topCard.topPill.icon
```
[the remark this answers is not in the document]
Comment (DwivediAman-exe) [reply Frontend/ui-common/src/Components/RideCompletedCard/View.purs:116]: > raise a PR for assetStore for assets

there will be apk update :(

File: Frontend/android-native/mobility-app/src/main/res/drawable/rectangle_9506.xml
[the remark this answers is not in the document]
Comment (DwivediAman-exe) [reply Frontend/android-native/mobility-app/src/main/res/drawable/rectangle_9506.xml:1]: I didn't add this image :(

**Review by ashishsingh101 — DISMISSED** [review]

**Comment by DwivediAman-exe** [comment]
This commit was used somewhere else.

**Pull request closed without merging**